test/cql-pytest/test_compaction_tombstone_gc.py

```diff
@@ -0,0 +1,141 @@
+
+            # let gc_grace_seconds and ttl pass
+            time.sleep(2)
+            # Insert backdated delete into the memtable.  It should not inhibit tombstone_gc
+            cql.execute(f"delete from {table} using timestamp 2 where k=1 and v=3")
+            # The following insert should not inhibit tombstone_gc since its timestamp is greater than the tombstones
+            cql.execute(f"insert into {table} (k, v) values (1, 4) using timestamp 5")
+            # do not flush before major compaction
+            nodetool.compact(cql, table, flush_memtables=False)
+
+            res = cql.execute(f"select * from mutation_fragments({table})")
+            sstables = set()
+            keys = set()
+            rows = set()
+            for r in res:
+                if "sstable" in r.mutation_source:
+                    print(r)
+                    if r.mutation_fragment_kind == "partition start":
+                        sstables.add(r.mutation_source)
+                        keys.add(r.k)
+                    elif r.mutation_fragment_kind == "clustering row":
+                        rows.add(r.v)
+                        assert not "tombstone" in r.metadata
+
+            assert len(sstables) == 1, f"Expected single sstable but saw {len(sstables)}: res={list(res)}"
+            assert keys == {1}, f"Expected keys=={1} but got {keys}: res={list(res)}"
+            assert rows == {3}, f"Expected rows=={3} but got {rows}: res={list(res)}"
+
+def test_tombstone_gc_with_materialized_view_update_in_memtable(scylla_only, cql, test_keyspace):
+    """
+    Reproduce https://github.com/scylladb/scylladb/issues/20424
+    """
+    schema = "k int primary key, v int, w int"
+    with new_test_table(cql, test_keyspace, schema) as table:
+        # Create a materialized view with same partition key as the base, and using a static column in the base as a clustering key in the view
+        with new_materialized_view(cql, table, '*', 'k, v', 'k is not null and v is not null', extra="with gc_grace_seconds = 0") as mv:
+            with nodetool.no_autocompaction_context(cql, mv):
+                # nodetool.setlogginglevel(cql, "compaction", "trace")
+                # Insert initial data into the base table
+                cql.execute(f"insert into {table} (k, v, w) values (1, 1, 1)")
+                # Flush the memtable so the following update won't get compacted in the memtable
+                nodetool.flush_keyspace(cql, test_keyspace)
+                # Update the static column in the view, causing a view update
+                # with a shadowable row tombstone for the old value and recent row_marker for the new value
+                cql.execute(f"insert into {table} (k, v) values (1, 2)")
+                # Flush all tables explicitly now, since this is skipped in the test on purpose by next major compaction.
+                nodetool.flush_all(cql)
+
+                # let gc_grace_seconds pass
+                time.sleep(1)
```

Comment:
Again, is this sleep() really needed? Doesn't gc_grace_seconds pass after 0 seconds, not 1?
All these sleep()s add up :-(

Reply:
Yes, it is needed.
I guess it's not exactly due to gc_grace_seconds but rather due to the commitlog min_gc time.
I'll fix the respective comments

Reply:
What is "commitlog min_gc time", where is it configured and why does it even exist? Is it documented anywhere?

Also, doesn't flush_all() flush the commitlog? If it does, how does anything related to commitlog have any effect?

Reply:
See https://github.com/scylladb/scylladb/pull/15060 by @elcallio 

And yes, flush_all does "flush" the commitlog in the sense that a new commitlog segment is forced.
However, writing stuff to the memtable writes also to the commitlog, and it will have new `min_gc_time`.
That said, this new min_gc_time is greater than the tombstone deletion time so it shouldn't interfere with tombstone garbage collection in theory.